packages/p2p/stream.go

```diff
@@ -54,38 +54,38 @@ func (m *Manager) dialPeer(ctx context.Context, p *peer.Peer, opts []ConnectPeer
 	if err != nil {
 		return nil, err
 	}
-	m.Libp2pHost.Peerstore().AddAddr(libp2pID, address, peerstore.ConnectedAddrTTL)
+	m.libp2pHost.Peerstore().AddAddr(libp2pID, address, peerstore.ConnectedAddrTTL)
 
 	if conf.useDefaultTimeout {
 		var cancel context.CancelFunc
 		ctx, cancel = context.WithTimeout(ctx, defaultConnectionTimeout)
 		defer cancel()
 	}
 
-	stream, err := m.Libp2pHost.NewStream(ctx, libp2pID, protocolID)
-	if err != nil {
-		return nil, errors.Wrapf(err, "dial %s / %s failed", address, p.ID())
-	}
-	ps := newPacketsStream(stream)
-	if err := sendNegotiationBlock(ps); err != nil {
-		err = errors.Wrap(err, "failed to send negotiation block")
-		err = errors.CombineErrors(err, stream.Close())
-		return nil, err
+	streams := make(map[protocol.ID]*PacketsStream)
+	for protocolID, handlers := range m.registeredProtocols {
+		stream, err := handlers.StreamEstablishFunc(ctx, libp2pID)
+		if err != nil {
+			return nil, errors.Wrapf(err, "dial %s / %s failed", address, p.ID())
+		}
+		m.log.Debugw("outgoing connection established",
+			"id", p.ID(),
+			"addr", stream.Conn().RemoteMultiaddr(),
+			"proto", protocolID,
+		)
+		streams[protocolID] = stream
 	}
-	m.log.Debugw("outgoing connection established",
-		"id", p.ID(),
-		"addr", stream.Conn().RemoteMultiaddr(),
-	)
-	return ps, nil
+
+	return streams, err
 }
 
-func (m *Manager) acceptPeer(ctx context.Context, p *peer.Peer, opts []ConnectPeerOption) (*packetsStream, error) {
+func (m *Manager) acceptPeer(ctx context.Context, p *peer.Peer, opts []ConnectPeerOption) (map[protocol.ID]*PacketsStream, error) {
```

Comment:
How does `acceptPeer` work with multiple protocols? In `dialPeer` we're calling the `handlers.StreamEstablishFunc` multiple times. But here it looks like we're just doing it once. Are we calling `acceptPeer` multiple times then from the outside?